### PR TITLE
Remove guaranteed residential solar, add random residential solar

### PR DESCRIPTION
**Pull request by Maleclypse**
#### Summary
Balance "Remove guaranteed residential solar, add random residential solar"

#### Purpose of change
Some house designs had guaranteed solar panels on the roof.  No more.  Adjust solar panel installations to Massachusetts ratio.

#### Describe the solution
I went and looked up the total number of single family homes in MA and then looked up the total number of residential solar installations in MA. 1,515,125 vs 177,143  This comes to 11.7%.  Since the game covers all of New England and I didn't want to do weights in fractions of a percent I averaged most houses to have an 11% chance of spawning a solar panel nest.  I also removed all guaranteed solar panel spawns on housetops.  

#### Describe alternatives you've considered
Nope

### CODE REVIEW

**Comment by GuardianDll** [comment]
is there a reason 3x3 solar is used exactly once?
is there a reason to make `[ "residential_1x2_solar", 6 ], [ "residential_2x2_solar", 2 ], [ "residential_1x3_solar", 3 ], [ "null", 88 ]` 100 times, instead of making a single nest with it
and, lastly, is there a reason to not put it in the roof_palette instead of adding it manually?

i'll fix it, because i feel it would be better this way

**Comment by Maleclypse** [comment]
> is there a reason 3x3 solar is used exactly once? is there a reason to make `[ "residential_1x2_solar", 6 ], [ "residential_2x2_solar", 2 ], [ "residential_1x3_solar", 3 ], [ "null", 88 ]` 100 times, instead of making a single nest with it and, lastly, is there a reason to not put it in the roof_palette instead of adding it manually?
> 
> i'll fix it, because i feel it would be better this way
> 
> UPD: only the first question is what i would like to know, if possible

So the 3x3 is replacing a 3x3 spawn on exactly one fairly large house.  I was expecting there to be multiple houses with 3x3 spawns when going through but only found one and a 4x3 that really didn't seem to fit the size of the house.   The answer to the rest of your questions is "I often stop looking for better ways to do a thing once I've found a way to do it, because this is how I avoid getting locked in decision paralysis of weighing the options." :)  Thank you for improving this.

**Comment by ehughsbaird** [comment]
I looked at this a while ago and got some different stats, from https://www.eia.gov/todayinenergy/detail.php?id=54379
- 4.7% of Northeast homes have solar
- 2.5% of commercial buildings

But the reason I was looking at it was because the monocrystalline solar panels (solar_panel_v2) only spawn one place in game, on a car. So I found some stats on the production of those year by year and found that if the solar added by year matched the ratios of monocrystalline/polycrystalline production (and assuming solar from before stats start is negligible), there'd be about 34% monocrystalline panels by power production (but it takes more polycrystalline panels to produce the same amount as a monocrystalline panel, but the stats are for production of panels, so...).


| Year  |  Solar Added (MW) | Monosilicon (%) | Added Monosilicon (MW) |
|-------|-------------------|-----------------|------------------------|
| 2011  | 42.65             | 25              | 10.66                  |
| 2012  | 134.5             | 25              | 33.63                  |
| 2013  | 226.94            | 25              | 56.74                  |
| 2014  | 269.11            | 25              | 67.28                  |
| 2015  | 286.53            | 25              | 71.63                  |
| 2016  | 394               | 25              | 98.5                   |
| 2017  | 571               | 35              | 199.85                 |
| 2018  | 370               | 45              | 166.5                  |
| 2019  | 237               | 65              | 154.05                 |
| Total | 2531.73           | -               | 858.83                 |

858.83/2531.73 = 34% high-efficiency monosilicon panels

Production Percent from https://www.ise.fraunhofer.de/content/dam/ise/de/documents/publications/studies/Photovoltaics-Report.pdf
Solar Added from https://www.mass.gov/info-details/renewable-energy-snapshot#installed-solar-capacity-in-massachusetts-

It's obviously not great data and I don't won't to add more work to you, but if you're interested it seems like it could fit in this PR.

**Comment by Maleclypse** [comment]
> I looked at this a while ago and got some different stats, from https://www.eia.gov/todayinenergy/detail.php?id=54379
> 
>     * 4.7% of Northeast homes have solar
> 
>     * 2.5% of commercial buildings
> 
> 
> But the reason I was looking at it was because the monocrystalline solar panels (solar_panel_v2) only spawn one place in game, on a car. So I found some stats on the production of those year by year and found that if the solar added by year matched the ratios of monocrystalline/polycrystalline production (and assuming solar from before stats start is negligible), there'd be about 34% monocrystalline panels by power production (but it takes more polycrystalline panels to produce the same amount as a monocrystalline panel, but the stats are for production of panels, so...).
> Year 	Solar Added (MW) 	Monosilicon (%) 	Added Monosilicon (MW)
> 2011 	42.65 	25 	10.66
> 2012 	134.5 	25 	33.63
> 2013 	226.94 	25 	56.74
> 2014 	269.11 	25 	67.28
> 2015 	286.53 	25 	71.63
> 2016 	394 	25 	98.5
> 2017 	571 	35 	199.85
> 2018 	370 	45 	166.5
> 2019 	237 	65 	154.05
> Total 	2531.73 	- 	858.83
> 
> 858.83/2531.73 = 34% high-efficiency monosilicon panels
> 
> Production Percent from https://www.ise.fraunhofer.de/content/dam/ise/de/documents/publications/studies/Photovoltaics-Report.pdf Solar Added from https://www.mass.gov/info-details/renewable-energy-snapshot#installed-solar-capacity-in-massachusetts-
> 
> It's obviously not great data and I don't won't to add more work to you, but if you're interested it seems like it could fit in this PR.

I'm guessing part of the difference in percentages comes from multi unit homes vs single unit homes.  Since in MA single family homes are about 1/2  57% per <https://framinghamsource.com/index.php/2022/01/16/57-2-of-massachusetts-homes-are-single-family-2nd-lowest-in-u-s/> the number of multi unit homes and there are few incentives for renters or landlords to add solar to a multifamily I think my math meshes with your math.   I'd have to create an advanced solar furniture which would be scope creep for this PR but definitely is something to consider for a followup.  This is already touching too many files in my opinion to add a different set of files to look at as well.  :)

File: data/json/mapgen_palettes/roof_palette.json [hunk 1]
```diff
@@ -55,6 +54,9 @@
     },
     "liquids": { "Я": { "liquid": "water_clean", "amount": [ 0, 300 ] } },
     "nested": {
+      "1": {
+        "chunks": [ [ "residential_1x2_solar", 6 ], [ "residential_2x2_solar", 2 ], [ "residential_1x3_solar", 3 ], [ "null", 88 ] ]
+      },
```
Comment (Maleclypse) [inline data/json/mapgen_palettes/roof_palette.json:59]:
```suggestion
      "1": {
        "chunks": [ [ "residential_1x2_solar", 6 ], [ "residential_2x2_solar", 2 ], [ "residential_1x3_solar", 3 ], [ "null", 88 ] ]
      },
      "2": {
        "chunks": [ [ "industrial_1x5_solar" ] ]
      },
```

File: data/json/mapgen_palettes/roof_palette.json [hunk 2]
```diff
@@ -55,6 +54,10 @@
...
+      "2": { "chunks": [ [ "industrial_1x5_solar" ] ] },
```
Comment (Maleclypse) [inline data/json/mapgen_palettes/roof_palette.json:60]:
```suggestion
      "2": { "chunks": [ [ "industrial_1x5_solar" ] ] },
      "3": { "chunks": [ [ "industrial_1x1_solar" ] ] },
```

File: data/json/mapgen_palettes/roof_palette.json [hunk 3]
```diff
@@ -55,6 +54,11 @@
...
```
Comment (Maleclypse) [inline data/json/mapgen_palettes/roof_palette.json:60]:
```suggestion
      "2": { "chunks": [ "industrial_1x5_solar" ] },
```

File: data/json/mapgen_palettes/roof_palette.json [hunk 4]
```diff
@@ -55,6 +54,11 @@
...
+      "3": { "chunks": [ [ "industrial_1x1_solar" ] ] },
```
Comment (Maleclypse) [inline data/json/mapgen_palettes/roof_palette.json:61]:
```suggestion
      "3": { "chunks": [ "industrial_1x1_solar" ] },
```

File: data/json/mapgen/nested/nested_chunks_roof.json [hunk 5]
```diff
@@ -531,5 +531,41 @@
...
+    "type": "mapgen",
+    "method": "json",
+    "nested_mapgen_id": "residential_3x3_solar",
+    "object": { "mapgensize": [ 3, 3 ], "place_furniture": [ { "furn": "f_solar_unit", "x": [ 0, 2 ], "y": [ 0, 2 ] } ] }
+  },
+  {
+    "type": "mapgen",
+    "method": "json",
+    "nested_mapgen_id": "industrial_1x5_solar",
+    "object": { "mapgensize": [ 1, 5 ], "place_furniture": [ { "furn": "f_solar_unit", "x": [ 0, 4 ], "y": 0 } ] }
```
Comment (Maleclypse) [inline data/json/mapgen/nested/nested_chunks_roof.json:563]:
```suggestion
    "object": { "mapgensize": [ 5, 5 ], "place_furniture": [ { "furn": "f_solar_unit", "x": [ 0, 4 ], "y": 0 } ] }
```

File: data/json/mapgen_palettes/roof_palette.json [hunk 6]
```diff
@@ -38,7 +38,6 @@
       "~": "f_chimney",
       "=": "f_vent_pipe",
       "A": "f_air_conditioner",
```
Comment (Maleclypse) [inline data/json/mapgen_palettes/roof_palette.json:40]:
```suggestion
      "A": "f_air_conditioner",
      "M": "f_solar_unit",
```

File: data/json/mapgen_palettes/roof_palette.json [hunk 7]
```diff
@@ -55,6 +55,11 @@
     },
     "liquids": { "Я": { "liquid": "water_clean", "amount": [ 0, 300 ] } },
     "nested": {
+      "1": {
+        "chunks": [ [ "residential_1x2_solar", 6 ], [ "residential_2x2_solar", 2 ], [ "residential_1x3_solar", 3 ], [ "null", 88 ] ]
+      },
+      "2": { "chunks": [ "industrial_1x5_solar" ] },
+      "3": { "chunks": [ "industrial_1x1_solar" ] },
```
Comment (Maleclypse) [inline data/json/mapgen_palettes/roof_palette.json:62]:
```suggestion
      "2": { "chunks": [ "industrial_1x1_solar" ] },
```

File: data/json/mapgen/irradiator_1.json [hunk 8]
```diff
@@ -420,8 +420,8 @@
         "    -..................................................RR..........-    ",
         "    -..................................................RR..........-    ",
         "    -..............................................................-    ",
-        "    -...::.......MMMMM.MMMMM.MMMMM.MMMMM...........X........X......-    ",
-        "    -...::.......MMMMM.MMMMM.MMMMM.MMMMM...........................-    ",
+        "    -...::.......2.....2.....2.....2...............X........X......-    ",
+        "    -...::.......2.....2.....2.....2...............................-    ",
```
Comment (Maleclypse) [inline data/json/mapgen/irradiator_1.json:424]:
```suggestion
        "    -...::.......22222.22222.22222.22222...........X........X......-    ",
        "    -...::.......22222.22222.22222.22222...........................-    ",
```

File: data/json/mapgen/shelter.json [hunk 9]
```diff
@@ -82,8 +82,8 @@
         "                        ",
         "                        ",
         "    ----------------    ",
-        "    -............M.-    ",
         "    -..............-    ",
+        "    -.........3....-    ",
```
Comment (Maleclypse) [inline data/json/mapgen/shelter.json:86]:
```suggestion
        "    -.........2....-    ",
```

File: data/json/mapgen/shelter.json [hunk 10]
```diff
@@ -164,9 +164,9 @@
         "    -.........v....-    ",
         "    -..............-    ",
         "    -.......&......-    ",
+        "    -..........3...-    ",
```
Comment (Maleclypse) [inline data/json/mapgen/shelter.json:167]:
```suggestion
        "    -..........2...-    ",
```

File: data/json/mapgen/shelter.json [hunk 11]
```diff
@@ -223,9 +223,9 @@
         "                        ",
         "           ----------   ",
         "   ---------........-   ",
-        "   -v.....M.........-   ",
+        "   -v...............-   ",
         "   -................-   ",
-        "  --................--  ",
+        "  --.....3..........--  ",
```
Comment (Maleclypse) [inline data/json/mapgen/shelter.json:228]:
```suggestion
        "  --.....2..........--  ",
```

File: data/json/mapgen/zoo.json [hunk 12]
```diff
@@ -276,7 +276,7 @@
         "                                                                                  -...........- ",
         "                                                                                  -...........- ",
         "                                                                -----             -...........- ",
-        "                                                                -...-             -......M....- ",
+        "                                                                -...-             -......3....- ",
```
Comment (Maleclypse) [inline data/json/mapgen/zoo.json:279]:
```suggestion
        "                                                                -...-             -......2....- ",
```

File: data/json/mapgen/zoo.json [hunk 13]
```diff
@@ -305,7 +305,7 @@
         "                                                                           -..................- ",
         "                                                                           -..................- ",
         "                                                                           -..................- ",
-        "                                                                           -...........M......- ",
+        "                                                                           -...........3......- ",
```
Comment (Maleclypse) [inline data/json/mapgen/zoo.json:308]:
```suggestion
        "                                                                           -...........2......- ",
```

**Pull request merged**